src/http/web_socket.cr

```diff
@@ -100,52 +101,109 @@ class HTTP::WebSocket
     @ws.close(message)
   end
 
-  def run
+  enum Opcode
+    Text
+    Binary
+    Close
+    Ping
+    Pong
+  end
+
+  struct Message
+    getter opcode
+
+    def text
+      @data.as(String)
+    end
+
+    def binary
+      @data.as(Bytes)
+    end
+
+    def initialize(@opcode : Opcode, @data : String | Bytes)
```

Comment:
`@data` could be left nilable.

```suggestion
    def initialize(@opcode : Opcode, @data : String | Bytes | Nil = nil)
```

Reply:
The `@data` should always have a value even if it's `""` so leaving it not-nilable.